### ISSUE TITLE
StridedSlice op support for DeepSpeech model

### ISSUE THREAD

**Issue by miaobin** [intel/webml-polyfill#1019]
StridedSlice in the DeepSpeech model:
![StridedSlice](https://user-images.githubusercontent.com/17840011/66628307-069cc200-ec31-11e9-9acd-16917aeae0f7.png)

  | NNAPI | ONNX | clDNN | MKLDNN | DML | MPS | BNNS
-- | -- | -- | -- | -- | -- | -- | --
op | [ANEURALNETWORKS_STRIDED_SLICE ](https://developer.android.com/ndk/reference/group/neural-networks#group___neural_networks_1ggaabbe492c60331b13038e39d4207940e0a89695302f8b1e7ae7ce8f4d8c0b8a752)|   |   |   |   | [MPSNDArrayStridedSlice ](https://developer.apple.com/documentation/metalperformanceshaders/mpsndarraystridedslice?language=objc)|  
inputs | inputs:     An n-D tensor, specifying the tensor to be sliced. |   |   |   |   |   |  
  | begin:        begin, a 1-D tensor of ANEURALNETWORKS_TENSOR_INT32. The starts of the   dimensions of the input tensor to be sliced. The length must be of   rank(input0). |   |   |   |   |   |  
  | end:        end, a 1-D tensor of ANEURALNETWORKS_TENSOR_INT32. The ends of the   dimensions of the input tensor to be sliced. The length must be of   rank(input0). |   |   |   |   |   |  
  | strides:        strides, a 1-D tensor of ANEURALNETWORKS_TENSOR_INT32. The strides of the   dimensions of the input tensor to be sliced. The length must be of   rank(input0). The entries must be non-zero. |   |   |   |   |   |  
  | begin_mask:        begin_mask, an ANEURALNETWORKS_INT32 scalar. If the ith bit of begin_mask   is set, begin[i] is ignored and the fullest possible range in that dimension   is used instead. |   |   |   |   |   |  
  | end_mask:        end_mask, an ANEURALNETWORKS_INT32 scalar. If the ith bit of end_mask is   set, end[i] is ignored and the fullest possible range in that dimension is   used instead. |   |   |   |   |   |  
  | shrink_axis_mask:        shrink_axis_mask, an ANEURALNETWORKS_INT32 scalar. If the ith bit of   shrink_axis_mask is set, the ith dimension specification shrinks the   dimensionality by 1, taking on the value at index begin[i]. In this case, the   ith specification must define a slice of size 1, e.g. begin[i] = x, end[i] =   x + 1. |   |   |   |   |   |  
outputs | outputs:     A tensor of the same OperandCode as input0 and rank (n - k), where k is the   number of bits set in shrink_axis_mask. For a   ANEURALNETWORKS_TENSOR_QUANT8_ASYMM tensor, the scale and zeroPoint must be   the same as input0. |   |   |   |   |   |  

Now, StridedSlice is working well with WebGL backend. And I am trying to support it with WASM backend. Because only four dimensions tensor is supported in tflite, I need do some workaround.

**Comment by miaobin** [comment]
I can only find Slice op form the website of [ONNX ](https://github.com/onnx/onnx/blob/master/docs/Operators.md#Slice)and [DML](https://docs.microsoft.com/en-us/windows/win32/api/directml/ns-directml-dml_slice_operator_desc).
@fujunwei @lisa0314 Please help fill in above table, thanks!

**Comment by lisa0314** [comment]
@miaobin BNNS has been updated. Thanks!

**Comment by ibelem** [comment]
@fujunwei Please update MPSNDArrayStridedSlice which is incorrect.

**Comment by NALLEIN** [comment]
Yes, it can extract a subset of source array using specified slice strides, but it's MPSKernel not MPSNNGraph node, it's a bit complicated to support in current design.

**Comment by fujunwei** [comment]
Yes, it can extract a subset of source array using specified slice strides, but it's MPSKernel not MPSNNGraph node, it's a bit complicated to support in current design.

**Comment by miaobin** [comment]
Update the table.

**Comment by pinzhenx** [comment]
MKLDNN updated. It can be implemented by primitive __reorder__ on submemory.

**Comment by ibelem** [comment]
@pinzhenx Good to know, thanks for the sharing! :) @lisa0314
